### PR TITLE
Create spiral.php

### PR DESCRIPTION
**Pull request by mattfanz**

### CODE REVIEW

File: week0002/mattfanz/spiral.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    if (!is_numeric($input) || $input <= 0) {
+        exit;
+    }
+}
+
+// MAIN
+$handleIn = fopen("php://stdin", "r");
+$handleOut = fopen('php://stdout', 'w');
+
+fwrite($handleOut, "Input N:");
```
Comment (vietlq) [inline week0002/mattfanz/spiral.php]: Please tell the use what M & N are. You can try this:

`Input the number of rows M:`

File: week0002/mattfanz/spiral.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
+<?php
+// FUNCTION
+function checkValidInput($input)
+{
+    if (!is_numeric($input) || $input <= 0) {
+        exit;
```
Comment (vietlq) [inline week0002/mattfanz/spiral.php]: You could print to STDERR before exiting to inform the user what went wrong.

**Comment by vietlq** [comment]
You also need to take care of spacing:

```
$ php week0002.php
Input N:4
Input M:4
1 2 3 4
12 13 14 5
11 16 15 6
10 9 8 7
```

File: week0002/mattfanz/spiral.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        fwrite(STDERR, "The input should be numberic and > 0".PHP_EOL);
+        exit;
+    }
+}
+
+// MAIN
+$handleIn = fopen("php://stdin", "r");
+$handleOut = fopen('php://stdout', 'w');
+
+fwrite($handleOut, "Input the number of rows N:");
```
Comment (vietlq) [inline week0002/mattfanz/spiral.php]: Ok so N is the number of rows or M is the number of rows? You need number of rows & number of columns.

File: week0002/mattfanz/spiral.php [hunk 4]
```diff
@@ -0,0 +1,72 @@
+<?php
+// FUNCTION
+function checkValidInput($input)
+{
+    if (!is_numeric($input) || $input <= 0) {
+        fwrite(STDERR, "The input should be numberic and > 0".PHP_EOL);
+        exit;
```
Comment (vietlq) [inline week0002/mattfanz/spiral.php]: Exit with code -1 should clearly indicate an error.

File: week0002/mattfanz/spiral.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+fscanf($handleIn, "%d", $n);
+checkValidInput($n);
+
+fwrite($handleOut, "Input the number of rows M:");
+fscanf($handleIn, "%d", $m);
+checkValidInput($m);
+
+$length = $n * $m;
+$left = 0;
+$right = $m-1;
```
Comment (vietlq) [inline week0002/mattfanz/spiral.php]: So `$m` is the number of columns, right?

Comment (mattfanz) [inline week0002/mattfanz/spiral.php]: yes $m is the number of columns

File: week0002/mattfanz/spiral.php (hunk 4 above)
Comment (mattfanz) [inline week0002/mattfanz/spiral.php]: Do you mean change to exit(-1) ?
I see on http://php.net/manual/en/function.exit.php write " Exit statuses should be in the range 0 to 254"

Comment (vietlq) [inline week0002/mattfanz/spiral.php]: Then make it 1 :)

**Comment by vietlq** [comment]
Please rename it php/week0002.php. Thank you!

**Pull request merged**